lib/spack/spack/hooks/cache_shell_script.py

```diff
@@ -29,7 +29,7 @@ def post_install(spec, explicit=None):
     for shell in shells_avail:
         mods = env_mods.shell_modifications(shell)
 
-        shell_script_path = os.path.join(spec.prefix, ".spack", f"{spec.name}_shell.{shell}")
+        shell_script_path = os.path.join(spec.prefix, ".spack", f"load.{shell}")
```

Comment:
note that this needs to be the same name that `load.py` later says to source -- maybe import the name from variable in spack.hooks.cache_shell_script

Reply:
Is this what you had in mind? [558aba7](https://github.com/spack/spack/pull/47755/commits/558aba76bf68af5ad7d051ec7a4852c20b547390)

Reply:
yep! that's better!